src/html/helpdata.cpp

```diff
@@ -339,9 +339,14 @@ inline static void CacheWriteString(wxOutputStream *f, const wxString& str)
 inline static wxString CacheReadString(wxInputStream *f)
 {
     size_t len = (size_t)CacheReadInt32(f);
-    wxCharBuffer str(len-1);
-    f->Read(str.data(), len);
-    return wxString(str, wxConvUTF8);
+    if (0<len)
```

Comment:
I don't think `wxCHECK` is appropriate, because condition failure doesn't indicate a programming error. But returning silently is wrong too, AFAICS from this (unfamiliar) code, size is really not supposed to be 0 here, so we should arguably log an error about the book being corrupted or something like that.